src/System.Private.CoreLib/shared/System/IO/BinaryReader.cs

```diff
@@ -484,15 +479,10 @@ private int InternalReadChars(Span<char> buffer)
                 // do ~1+log(n) reads to read n characters.
                 numBytes = charsRemaining;
 
-                // special case for DecoderNLS subclasses when there is a hanging byte from the previous loop
```

Comment:
This hack should not be needed (cleanup from CoreRT). I was not able to trace down why it was added.